pyanaconda/users.py

```diff
@@ -112,87 +112,122 @@ def cryptPassword(password, algo=None):
 
     return cryptpw
 
-def validatePassword(pw, user="root", settings=None, minlen=None, empty_ok=False):
+class PwqualitySettingsCache(object):
+    """Cache for libpwquality settings used for password validation.
+
+    Libpwquality settings instantiation is probably not exactly cheep
```

Comment:
This is in the context of computationally expensive/cheap, but I guess I should make that more evident.